ipam/manager_test.go

```diff
@@ -141,233 +141,279 @@ func cleanupTestAddressSpace(am AddressManager) error {
-func TestAddressPoolRequestsForSamePool(t *testing.T) {
-	// Start with the test address space.
-	am, err := createAddressManager()
-	if err != nil {
-		t.Fatalf("createAddressManager failed, err:%+v.", err)
-	}
-
-	// Request the same address pool twice.
-	poolId1, subnet1, err := am.RequestPool(LocalDefaultAddressSpaceId, "", "", nil, false)
-	if err != nil {
-		t.Errorf("RequestPool failed, err:%v", err)
-	}
-
-	poolId2, subnet2, err := am.RequestPool(LocalDefaultAddressSpaceId, poolId1, "", nil, false)
-	if err != nil {
-		t.Errorf("RequestPool failed, err:%v", err)
-	}
-
-	// Test the subnets do not match.
-	if poolId1 != poolId2 || subnet1 != subnet2 {
-		t.Errorf("Pool requests returned different pools.")
-	}
-
-	// Release the address pools.
-	err = am.ReleasePool(LocalDefaultAddressSpaceId, poolId1)
-	if err != nil {
-		t.Errorf("ReleasePool failed, err:%v", err)
-	}
-
-	err = am.ReleasePool(LocalDefaultAddressSpaceId, poolId2)
-	if err != nil {
-		t.Errorf("ReleasePool failed, err:%v", err)
-	}
-
-	// Third release should fail.
-	err = am.ReleasePool(LocalDefaultAddressSpaceId, poolId1)
-	if err == nil {
-		t.Errorf("ReleasePool succeeded extra, err:%v", err)
-	}
+func TestManager(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Manager Suite")
 }
 
-// Tests address requests from the same pool return separate addresses and releases work correctly.
-func TestAddressRequestsFromTheSamePool(t *testing.T) {
-	// Start with the test address space.
-	am, err := createAddressManager()
-	if err != nil {
-		t.Fatalf("createAddressManager failed, err:%+v.", err)
-	}
-
-	// Request a pool.
-	poolId, _, err := am.RequestPool(LocalDefaultAddressSpaceId, "", "", nil, false)
-	if err != nil {
-		t.Errorf("RequestPool failed, err:%v", err)
-	}
-
-	// Request two addresses from the pool.
-	address1, err := am.RequestAddress(LocalDefaultAddressSpaceId, poolId, "", nil)
-	if err != nil {
-		t.Errorf("RequestAddress failed, err:%v", err)
-	}
-
-	addr, _, _ := net.ParseCIDR(address1)
-	address1 = addr.String()
-
-	address2, err := am.RequestAddress(LocalDefaultAddressSpaceId, poolId, "", nil)
-	if err != nil {
-		t.Errorf("RequestAddress failed, err:%v", err)
-	}
-
-	addr, _, _ = net.ParseCIDR(address2)
-	address2 = addr.String()
-
-	// Test the addresses do not match.
-	if address1 == address2 {
-		t.Errorf("Address requests returned the same address %v.", address1)
-	}
-
-	// Release addresses and the pool.
-	err = am.ReleaseAddress(LocalDefaultAddressSpaceId, poolId, address1, nil)
-	if err != nil {
-		t.Errorf("ReleaseAddress failed, err:%v", err)
-	}
-
-	err = am.ReleaseAddress(LocalDefaultAddressSpaceId, poolId, address2, nil)
-	if err != nil {
-		t.Errorf("ReleaseAddress failed, err:%v", err)
-	}
-
-	err = am.ReleasePool(LocalDefaultAddressSpaceId, poolId)
-	if err != nil {
-		t.Errorf("ReleasePool failed, err:%v", err)
-	}
-}
+var (
```

Comment:
was unit test for requestpool, requestaddr, releaseddr, releasepool moved somewhere else?